[cutlass backend] make compile name independent of command

henrylhtsang opened the pull request:
Differential Revision: D74291603

The goal is to reuse the kernels as much as possible.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

ColinPeppler commented on torch/_inductor/codecache.py:
```diff
@@ -3363,8 +3363,22 @@ def write(cls, source_code: str, dst_file_ext: str) -> tuple[str, str]:
+                # cutlass key
+                cutlass_key(),
+            # hack to deal with AOTI .o compilation
+            ] + [dst_file_ext] if dst_file_ext == "o" else [])
```
why do you need this hack?

henrylhtsang replied:
> why do you need this hack?

@ColinPeppler I need dst_file_ext for the tests to pass. But I want to keep the source name when we (or you) eventually fix the .o compilation.

henrylhtsang committed:
[cutlass backend] make compile name independent of command (#153388)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/153388

Test Plan: ci

Reviewed By: ColinPeppler

Differential Revision: D74291603
